File: 3-string/c-normal-problem.py
def visualizarPorDentro(s: str) -> str:
    # Inverte a string
    s = s[::-1]
    
    # Troca p por q e vice-versa
    # Em Python, strings são imutáveis, então vamos criar uma nova string
    resultado = ""
    for c in s:
        if c == 'p':
            resultado += 'q'
        elif c == 'q':
            resultado += 'p'
        else:  # w permanece w
            resultado += c
            
    return resultado

def main():
    # Lê o número de casos de teste
    t = int(input())
    
    # Processa cada caso de teste
    for _ in range(t):
        s = input()
        print(visualizarPorDentro(s))

if __name__ == "__main__":
    main()

File: 3-string/c-normal-problem.cpp
#include <algorithm>
#include <iostream>
#include <string>
using namespace std;

// Funcao para inverter e trocar p por q e vice-versa
string visualizarPorDentro(string s) {
    // Inverte a string
    reverse(s.begin(), s.end());

    // Troca p por q e vice-versa
    for (char& c : s) {
        if (c == 'p')
            c = 'q';
        else if (c == 'q')
            c = 'p';
        // w permanece w
    }

    return s;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        string s;
        cin >> s;
        cout << visualizarPorDentro(s) << endl;
    }

    return 0;
}
